router/oauthResponseHandler/oauthResponseHandler.go

```diff
@@ -125,6 +129,9 @@ func (authErrHandler *OAuthErrResHandler) Setup() {
 	authErrHandler.destAuthInfoMap = make(map[string]*AuthResponse)
 	authErrHandler.refreshActiveMap = make(map[string]bool)
 	authErrHandler.disableDestActiveMap = make(map[string]bool)
+	if authErrHandler.workspaceToken == "" {
+		authErrHandler.workspaceToken = authErrHandler.TokenProvider.AccessToken()
+	}
```

Comment:
[minor] I believe, we don't need `workspaceToken` anymore, since we can call `TokenProvider` every time.
```suggestion
```